src/native/libs/System.Security.Cryptography.Native/pal_evp_mac.h

```diff
@@ -0,0 +1,186 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "pal_compiler.h"
+#include <stdint.h>
+#include "opensslshim.h"
+
+/*
+Shims the EVP_MAC_fetch function.
+
+algorithm: The name of the algorithm to fetch.
+haveFeature: A pointer to an int32_t. When this function returns, the value will
+             contain an integer to determine if the platform supports EVP_MAC_fetch.
+             0 indicates that the platform does not support EVP_MAC_fetch.
+             1 indicates that the platform does support EVP_MAC_fetch.
+
+return: A pointer to an EVP_MAC. This pointer may be NULL if OpenSSL failed to allocate internally,
+        or, if the platform does not support EVP_MAC_fetch. Use the haveFeature value to determine
+        if the NULL value is due to allocation failure or lack of platform support.
```

Comment:
Wouldn't it also fail if the algorithm is unknown (which could include being disabled in a private build, or "from the future")?